src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -8333,6 +8376,281 @@ private StatementSyntax ParseLocalDeclarationStatement()
+        /// </summary>
+        private VariableDeclarationSyntax TryParseDeconstructionDeclaration(bool withEquals)
+        {
+            if (this.CurrentToken.Kind == SyntaxKind.OpenParenToken
+               || (CurrentToken.IsVar() && this.PeekToken(1).Kind == SyntaxKind.OpenParenToken))
+            {
+                var resetPoint = this.GetResetPoint();
+
+                try
+                {
+                    var deconstruction = ParseDeconstructionDeclaration(withEquals);
+                    if (deconstruction == null || !DeconstructionVariableLooksGood(deconstruction, withEquals))
+                    {
+                        this.Reset(ref resetPoint);
+                        return null;
+                    }
+                    else
+                    {
+                        return deconstruction;
+                    }
+
+                }
+                finally
+                {
+                    this.Release(ref resetPoint);
+                }
+            }
+
+            return null;
+        }
+
+        /// <summary>
+        /// Parses a deconstruction-declaration, which can appear in a local-declaration statement or a for statement.
+        /// Returns null if this does not look like a deconstruction-declaration after all (equal sign missing).
+        ///
+        /// The syntax is either var form: `var (deconstruction-declaration, ...) = expression` or list form `(deconstruction-declaration, ...) = expression`.
+        /// Cannot return null, except at the top-level.
+        /// </summary>
+        /// <param name="withEquals">Specifies whether to look for and consume the equals sign and the following expression.</param>
+        /// <param name="topLevel">Specifies whether to parse the terminal form of a deconstruction-declaration (which can't appear at the top-level).</param>
+        private VariableDeclarationSyntax ParseDeconstructionDeclaration(bool withEquals, bool topLevel = true)
+        {
+            Debug.Assert(topLevel || !withEquals); // withEquals can only be set at the top-level
+
+            VariableDeclarationSyntax result;
+            if (this.CurrentToken.IsVar() && this.PeekToken(1).Kind == SyntaxKind.OpenParenToken)
+            {
+                // parses `var (...) = expression` form
+                result = ParseDeconstructionVarForm(withEquals);
+            }
+            else if (this.CurrentToken.Kind == SyntaxKind.OpenParenToken)
+            {
+                // parses `(...) = expression` form
+                result = ParseDeconstructionList(withEquals, justIdentifiers: false);
+            }
+            else if (!topLevel)
+            {
+                // parses `type id` part
+                result = ParseDeconstructionTypeIdPart();
+            }
+            else
+            {
+                return null;
+            }
+
+            return CheckFeatureAvailability(result, MessageID.IDS_FeatureTuples);
```

Comment:
Added negative test. Positive tests will be easier once I have some binding, so they will be added in subsequent PRs.